scilpy/utils/bvec_bval_tools.py

```diff
@@ -120,6 +120,70 @@ def get_shell_indices(bvals, shell, tol=10):
         np.logical_and(bvals < shell + tol, bvals > shell - tol))[0]
 
 
+def volumes_iterator(img, size, dtype=None):
+    """Generator that iterates on gradient volumes of data"""
+
+    nb_volumes = img.shape[-1]
+
+    if size == nb_volumes:
+        yield list(range(nb_volumes)), img.get_data(dtype=dtype)
```

Comment:
`get_data` -> `get_fdata`

Also, this line is redundant with line 169, I would remove either one of them and handle the block size accordingly.

Reply:
I agree that's bad. I didn't want to change too much Philippe's work so I copied and paste. I'll try to change that.

You say "as it was before". Where?

Reply:
> You say "as it was before". Where? 
I just meant that before the split this line wasn't there and the block size was just set to `img.shape[-1]` if it wasn't specified in args